### PR TITLE
Add nats nats-server and nsc.

### PR DESCRIPTION
**Pull request by dlorenc**
These are all required to build the full nats image.

Fixes: 

Related: 

#### For new package PRs only 

- [x] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [x] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [x] REQUIRED - The version of the package is still receiving security updates

### CODE REVIEW

File: nats-server.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+pipeline:
+  # We can't use go/install because this requires a specific go version
+  - uses: git-checkout
+    with:
+      repository: https://github.com/nats-io/nats-server
+      tag: v${{package.version}}
+      expected-commit: b91fa85462d42c2f988170aee27955773e68c56d
+  - runs: |
+      mkdir -p ${{targets.destdir}}/usr/bin
+      mkdir -p ${{targets.destdir}}/nats/conf
```
Comment (jdolitsky) [inline nats-server.yaml:24]: /nats ? or should it be something like /etc/nats ?

Comment (dlorenc) [reply nats-server.yaml:24]: This is the path used in the upstream docker image: https://github.com/nats-io/nats-server/blob/main/docker/Dockerfile.nightly

I'm not sure why, I'll have to dig in to see if it's a hardcoded default. It's definitely weird.

Comment (jdolitsky) [reply nats-server.yaml:24]: sorry confused - are we saying switch to /etc or keep?

Comment (jdolitsky) [reply nats-server.yaml:24]: (it looks like still in the root)

Comment (dlorenc) [reply nats-server.yaml:24]: d'oh, never pushed the branch. nice catch.

**Review by jdolitsky — APPROVED** [review]

**Pull request merged**